src/validators/dataclass.rs

```diff
@@ -176,6 +178,28 @@ impl Validator for DataclassArgsValidator {
                     ($args:ident, $get_method:ident, $get_macro:ident, $slice_macro:ident) => {{
                         // go through fields getting the value from args or kwargs and validating it
                         for (index, field) in self.fields.iter().enumerate() {
+                            if (!field.init) {
+                                match field.validator.default_value(py, Some(field.name.as_str()), state) {
+                                    Ok(Some(value)) => {
+                                        // Default value exists, and passed validation if required
+                                        set_item!(field, value);
+                                    },
+                                    Ok(None) => continue,
+                                    Err(ValError::Omit) => continue,
```

Comment:
```suggestion
                                    Ok(None) | Err(ValError::Omit) => continue,
```

Arguably the `continue` is also not necessary because the block immediately has `continue` after the `match`, so this could just be `=> {},`